### ISSUE TITLE
arm-deploy Action failing after Release 2.24.1

### ISSUE THREAD

**Issue by t-dedah** [Azure/azure-cli#18360]
Arm deployment using [arm-deploy action](https://github.com/Azure/arm-deploy) started failing after Release 2.24.1 with error:-

```
##[debug]Exit code 0 received from tool '/usr/bin/az'
##[debug]STDIO streams have closed for tool '/usr/bin/az'
Error: The process '/usr/bin/az' failed because one or more lines were written to the STDERR stream
```

Update:
The data we are getting in stderr is a string with 71 spaces so just a blank line which is forcing the action to fail.

**Comment by yonzhan** [comment]
ARM

**Comment by zhoxing-ms** [comment]
1. Could you please show me the specific error message?
2. In order to fix a possible related issue https://github.com/Azure/azure-cli/issues/18276 , the CLI recently upgraded to version `2.24.2`. Could you update the CLI to the latest version and try again?

**Comment by t-dedah** [comment]
@zhoxing-ms The error string is just a whitespace string. But as its length is 71, the actions fails as failOnError is set to true.

```
##[debug]Evaluating condition for step: 'Run azure/arm-deploy@v1'
##[debug]Evaluating: success()
##[debug]Evaluating success:
##[debug]=> true
##[debug]Result: true
##[debug]Starting: Run azure/arm-deploy@v1
##[debug]Loading inputs
##[debug]Loading env
Run azure/arm-deploy@v1
Validating template...
Warning:                                                                       
Creating deployment...
##[debug]exec tool: /usr/bin/az
##[debug]arguments:
##[debug]   deployment
##[debug]   group
##[debug]   create
##[debug]   --resource-group
##[debug]   deepak-rg
##[debug]   --template-file
##[debug]   ./azuredeploy.json
##[debug]   --mode
##[debug]   Incremental
##[debug]   --parameters
##[debug]   ./azuredeploy.parameters.json
##[debug]   -o
##[debug]   json
Error:                                                                       
Error:                                                                       
##[debug]Exit code 0 received from tool '/usr/bin/az'
##[debug]STDIO streams have closed for tool '/usr/bin/az'
Error: The process '/usr/bin/az' failed because one or more lines were written to the STDERR stream
##[debug]Node Action run completed with exit code 1
##[debug]Finishing: Run azure/arm-deploy@v1
```

**Comment by zhoxing-ms** [comment]
Could you please send the specific template file and parameter file to me by email? My email address is Zhou.Xing@microsoft.com

**Comment by bradystroud** [comment]
I am having the same problem

**Comment by t-dedah** [comment]
@zhoxing-ms Emailed you the templates but as we are using sample templates they aren't anything special. Also, everyone is facing this issue irrespective of the template as discussed here https://github.com/Azure/arm-deploy/issues/48

**Comment by zhoxing-ms** [comment]
ARM deployment itself is normal and successful, this issue should be related to the output stream. It's root cause looks the same as issue #18262

**Comment by yonzhan** [comment]
Add @Juliehzl for awareness

**Comment by cmatskas** [comment]
I can confirm that a standard ARM deployment is also failing for me with the same error message. I can successfully run the ARM deployment with the same Service Principal locally. Is there an update or a workaround for this?

**Comment by bradystroud** [comment]
> I can confirm that a standard ARM deployment is also failing for me with the same error message. I can successfully run the ARM deployment with the same Service Principal locally. Is there an update or a workaround for this?

See https://github.com/Azure/arm-deploy/issues/48

**Comment by daveabrock** [comment]
I know this is a Closed issue, but I want to pass along some feedback (and I've relayed this to our Microsoft contacts as well). It was still fixed rather quickly, which was appreciated, but this caused significant issues with our customers in the last few weeks (we did back out the update). I object to this issue being flagged as a `feature-request` when customers are broken (and "failing" is in the issue title). I'm not sure if flagging it as a bug would trigger better communication/reporting with customers, but I wanted to provide the feedback.
